cloud/google/cmd/gce-machine-controller/app/options/options.go

```diff
@@ -37,6 +40,9 @@ func NewMachineControllerServer() *MachineControllerServer {
 func (s *MachineControllerServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.KubeadmToken, "token", s.KubeadmToken, "Kubeadm token to use to join new machines")
 	fs.StringVar(&s.MachineSetupConfigsPath, "machinesetup", s.MachineSetupConfigsPath, "path to machine setup configs file")
+	fs.StringVar(&s.SSHPrivateKeyPath, "sshprivatekeypath", s.SSHPrivateKeyPath, "path to the private ssh key file")
+	fs.StringVar(&s.SSHPublicKeyPath, "sshpublickeypath", s.SSHPrivateKeyPath, "path to the public ssh key file")
+	fs.StringVar(&s.SSHUserPath, "sshuserpath", s.SSHUserPath, "path to the ssh user file")
```

Comment:
Because the user is in a secret. I found how to get secrets into mounted paths and into environment variables but did not see an obvious way to get it directly into a pod argument. If you know how, could you also provide a pointer.

Reply:
It's combination of pulling the secret into an env var, and then pulling the env var into the container args. https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#use-environment-variables-to-define-arguments